cmd/root.go

```diff
@@ -260,5 +250,41 @@ func (c *rootCommand) setupLoggers() (<-chan struct{}, error) {
 		})
 		c.globalState.Logger.Debug("Logger format: TEXT")
 	}
-	return ch, nil
+
+	cancel := func() {} // noop as default
+	if hook != nil {
+		ctx := context.Background()
+		ctx, cancel = context.WithCancel(ctx)
+		c.setLoggerHook(ctx, hook)
+	}
+
+	// Sometimes the Go runtime uses the standard log output to
+	// log some messages directly.
+	// It does when an invalid char is found in a Cookie.
+	// Check for details https://github.com/grafana/k6/issues/711#issue-341414887
+	w := c.globalState.Logger.Writer()
+	stdlog.SetOutput(w)
+	c.loggersWg.Add(1)
+	go func() {
+		<-stop
+		cancel()
+		_ = w.Close()
+		c.loggersWg.Done()
+	}()
+	return nil
+}
+
+type asyncHook interface {
+	logrus.Hook
+	Listen(ctx context.Context)
+}
```

Comment:
Could you move this to the `log` package, and export it there?

That way you don't have to export `LokiHook`, and it would be less changes overall, I think.

Plus, this is really a logging interface, so it belongs more in `log` than in `cmd`.